packages/valory/skills/keep3r_job_abci/behaviours.py

```diff
@@ -1092,29 +1175,52 @@ def get_tx(self) -> Generator[None, None, str]:
         """
         keeper_address = self.synchronized_data.safe_contract_address
         bonding_asset = self.context.params.k3pr_address
-        k3pr_amount = yield from self.get_pending_unbonds(keeper_address, bonding_asset)
+        k3pr_amount = yield from self.withdrawn_funds(keeper_address, bonding_asset)
         if k3pr_amount is None:
             # something went wrong
             return SwapAndDisburseRewardsRound.ERROR_PAYLOAD
 
+        multisend_txs: List[Dict[str, Any]] = []
+        # check if we already withdrew the funds
+        # if we did, we just need to swap the k3pr
+        swap_already_withdrawn_funds = yield from self.swap_already_withdrawn_funds(
+            keeper_address, bonding_asset
+        )
+        if not swap_already_withdrawn_funds:
+            # 1. get the withdraw transaction
+            pending_unbonds = yield from self.get_pending_unbonds(
+                keeper_address, bonding_asset
+            )
+            if pending_unbonds is None:
+                # something went wrong
+                return SwapAndDisburseRewardsRound.ERROR_PAYLOAD
+            k3pr_amount += pending_unbonds
+            withdraw_tx = yield from self._get_withdraw_tx(
+                self.keep3r_v2_contract_address, bonding_asset
+            )
+            if withdraw_tx is None:
+                # something went wrong
+                return SwapAndDisburseRewardsRound.ERROR_PAYLOAD
+            multisend_txs.append(withdraw_tx)
```

Comment:
This is the modified logic for checking whether we need to swap the k3pr already in the safe